From 97e90af3168f4997fd96c391ae1ee021bf39ba5f Mon Sep 17 00:00:00 2001
From: Andy CrossGate Yan <GeForce8800Ultra@gmail.com>
Date: Tue, 11 Aug 2026 15:02:49 +0000
Subject: [PATCH] Revert "Biometrics: Allow disabling of fingerprint cleanups"

This reverts commit add9f4dc93bba9d414ad67a0770b82c06747d09f.

[Port to A16] LineageOS moved the config_cleanupUnusedFingerprints gate from the
deleted HIDL Fingerprint21 to the AIDL FingerprintProvider (de-HIDL refactor
0d94465a0c22); the same three sites (field, constructor read,
scheduleInternalCleanup early-return) are reverted at the new location.

Change-Id: I6421d0009e7cf1d9580cc45f98e9c1e4d17acf2a
---
 .../sensors/fingerprint/aidl/FingerprintProvider.java     | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/services/core/java/com/android/server/biometrics/sensors/fingerprint/aidl/FingerprintProvider.java b/services/core/java/com/android/server/biometrics/sensors/fingerprint/aidl/FingerprintProvider.java
index 51ca147fc532..48d6a7df76a6 100644
--- a/services/core/java/com/android/server/biometrics/sensors/fingerprint/aidl/FingerprintProvider.java
+++ b/services/core/java/com/android/server/biometrics/sensors/fingerprint/aidl/FingerprintProvider.java
@@ -136,8 +136,6 @@ public class FingerprintProvider implements IBinder.DeathRecipient, ServiceProvi
     @Nullable private IVirtualHal mVhal;
     @Nullable private String mHalInstanceNameCurrent;
 
-    private boolean mCleanup;
-
     private final class BiometricTaskStackListener extends TaskStackListener {
         @Override
         public void onTaskStackChanged() {
@@ -209,9 +207,6 @@ public class FingerprintProvider implements IBinder.DeathRecipient, ServiceProvi
         mTestHalEnabled = testHalEnabled;
         mBiometricHandlerProvider = biometricHandlerProvider;
 
-        mCleanup = context.getResources().getBoolean(
-                org.lineageos.platform.internal.R.bool.config_cleanupUnusedFingerprints);
-
         initAuthenticationBroadcastReceiver();
         initFingerprintDanglingBroadcastReceiver();
         initSensors(resetLockoutRequiresHardwareAuthToken, props, gestureAvailabilityDispatcher);
@@ -684,9 +679,6 @@ public class FingerprintProvider implements IBinder.DeathRecipient, ServiceProvi
     @Override
     public void scheduleInternalCleanup(int sensorId, int userId,
             @Nullable ClientMonitorCallback callback, boolean favorHalEnrollments) {
-        if (!mCleanup) {
-            return;
-        }
         mHandler.post(() -> {
             final FingerprintInternalCleanupClient client =
                     new FingerprintInternalCleanupClient(mContext,
-- 
2.43.0

